classes/Project/Project.mjs

```diff
@@ -1,33 +1,36 @@
 import dbDriver from "../../database/driver.mjs"
-import { validateProjectPayload } from "../../utilities/validatePayload.mjs"
+import {validateProjectPayload} from "../../utilities/validatePayload.mjs"
 import {User} from "../User/User.mjs"
 
 let err_out = Object.assign(new Error(), {
   status: 500,
-  message: "Unknown Server error",
- })
+  message: "Unknown Server error"
+})
 
 const database = new dbDriver("mongo")
 
 export default class Project {
+  #creator
   constructor(userId) {
-    this.creator = userId
+    this.id = userId
   }
 
+  set creator(userAgent) {
+    this.#creator = userAgent
+  }
   async create(payload) {
     // validation checks for all the required elements without which a project cannot be created. modify validateProjectPayload function to include more elements as they become available (layers,... )
     const validation = validateProjectPayload(payload)
- 
+
     if (!validation.isValid) {
       err_out.status = 400
-      err_out.message =   validation.errors
+      err_out.message = validation.errors
       throw err_out
     }
 
     try {
-     return await database.save({...payload, "@type": "Project"})
-    
-    } catch (err) { 
+      return await database.save({...payload, "@type": "Project"})
```

Comment:
I would rather the driver handle a `saveProject(payload)` than hardcode the `@type` convenience in here.